src/merkleblock.cpp

```diff
@@ -49,6 +54,21 @@ CMerkleBlock::CMerkleBlock(const CBlock& block, CBloomFilter* filter, const std:
             vMatch.push_back(false);
         }
         vHashes.push_back(hash);
+        if (prove_witness && i) {
```

Comment:
3abfd6d3d99bbcceb09162e83d8f9f9769bd4036:

```cpp
if (prove_witness) {
  if (i == 0) {
    // generation tx has null wtxid
    wtxids.emplace_back();
  } else {
    wtxids.push_back(block.vtx[i]->GetWitnessHash());
  }
```

Reply:
The null wtxid is already added at the top outside the loop.